packages/n-back/src/index.ts

```diff
@@ -0,0 +1,105 @@
+        trial_sequence.push(trial_sequence[i - n]);
+    } else {
+        const possible_stimuli = stimuli.filter(function (s: any) {
+          return (i < n || s !== trial_sequence[i - n]);
+        });
+        const random_stimulus = jsPsych.randomization.sampleWithoutReplacement(possible_stimuli, 1)[0];
+        trial_sequence.push(random_stimulus)
+      }
+    }
+
+  const timeline: any[] = [];
+
+  for (var i = 0; i < trial_sequence.length; i++) {
+
+    timeline.push({
+      type: jsPsychHtmlKeyboardResponse,
+      stimulus: `<p style="font-size: 48px; color: gray;">+</p>`,
+      choices: "NO_KEYS",
+      trial_duration: fixation_duration
+    });
+
+    timeline.push({
+      type: jsPsychHtmlKeyboardResponse,
+      stimulus: `<p style="font-size: 48px;">${trial_sequence[i]}</p>`,
+      choices: [keyboard_response],
+      trial_duration: trial_duration,
+      post_trial_gap: post_trial_gap,
+      data: { correct: i >= 2 && trial_sequence[i] === trial_sequence[i - n] },
+      on_finish: function (data: any) {
+        data.correct_response = data.correct && data.response === keyboard_response;
+        data.correct_no_response = !data.correct && data.response === null;
+      }
+    })
+  }
+
+  if (debrief){
+  if (return_accuracy){
+    timeline.push(
+      {
+        type: jsPsychHtmlKeyboardResponse,
+        stimulus: function(){
+          var correct_responses = jsPsych.data.get().filter({correct_response: true}).count();
+          var correct_no_responses = jsPsych.data.get().filter({correct_no_response: true}).count();
+          var total_trials = jsPsych.data.get().count();
+          var accuracy = Math.round(((correct_responses + correct_no_responses) / total_trials) * 100);
+          return `<p>Thank you for participating!</p>
+            <p>You correctly responded to <strong>${correct_responses}</strong> matching trials.</p>
+            <p>You correctly not responded to <strong>${correct_no_responses}</strong> non-matching trials.</p>
+            <p>Your accuracy was <strong>${accuracy}%</strong>.</p>
+            <p>Press any key to finish the experiment.</p>`;
+        },
+        choices: "NO_KEYS",
+        on_start: function () {
+          if (data_output == "csv"){
+          jsPsych.data.get().localSave('csv', `n_back.csv`);} else if (data_output == "json") {
+            jsPsych.data.get().localSave('json', `n_back.json`);
+          }
+        },
+        simulation_options: {
+          simulate: false
+        }
+      })
+  } else {
+    timeline.push(
+      {
+        type: jsPsychHtmlKeyboardResponse,
+        stimulus: function(){
+          return `<p>Thank you for participating!</p>
+            <p>Press any key to finish the experiment.</p>`;
+        },
+        choices: "NO_KEYS",
+        on_start: function () {
+          if (data_output == "csv"){
+            jsPsych.data.get().localSave('csv', `n_back.csv`);} else if (data_output == "json") {
```

Comment:
just wondering, is it normal to `.localSave()`, or can we give the developer the option to locally save or not?

Reply:
Implemented the option